src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCommitter.cs

```diff
@@ -203,21 +202,21 @@ private Solution CreateSolutionWithOriginalName(Document document, CancellationT
                 return solution;
             }
 
-            private bool TryGetSymbol(Solution solutionWithOriginalName, DocumentId documentId, CancellationToken cancellationToken, out ISymbol symbol)
+            private async Task<ISymbol> TryGetSymbolAsync(Solution solutionWithOriginalName, DocumentId documentId, CancellationToken cancellationToken)
             {
                 var documentWithOriginalName = solutionWithOriginalName.GetDocument(documentId);
-                var syntaxTreeWithOriginalName = documentWithOriginalName.GetSyntaxTreeAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                var syntaxTreeWithOriginalName = await documentWithOriginalName.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
 
                 var syntaxFacts = documentWithOriginalName.GetLanguageService<ISyntaxFactsService>();
                 var semanticFacts = documentWithOriginalName.GetLanguageService<ISemanticFactsService>();
-                var semanticModel = documentWithOriginalName.GetSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                var semanticModel = await documentWithOriginalName.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
                 var token = syntaxTreeWithOriginalName.GetTouchingWord(_snapshotSpan.Start, syntaxFacts, cancellationToken);
 
-                symbol = semanticFacts.GetDeclaredSymbol(semanticModel, token, cancellationToken);
+                var symbol = semanticFacts.GetDeclaredSymbol(semanticModel, token, cancellationToken);
                 symbol = symbol ?? semanticModel.GetSymbolInfo(token, cancellationToken).Symbol;
 
-                return symbol != null;
+                return symbol;
```

Comment:
Consider merging the last 3 lines to `return semanticFacts.GetDeclaredSymbol(semanticModel, token, cancellationToken) ?? semanticModel.GetSymbolInfo(token, cancellationToken).Symbol` (with a line break if necessary).